packages/flutter_tools/lib/src/compile.dart

```diff
@@ -12,8 +12,12 @@ import 'base/io.dart';
 import 'base/process_manager.dart';
 import 'globals.dart';
 
+typedef void CompilerMessageConsumer(String message);
+
 class _StdoutHandler {
-  _StdoutHandler() {
+  final CompilerMessageConsumer consumer;
```

Comment:
s/consumer/errorHandler/ ?

Reply:
It consumes all output (stdout) which might not necessarily errors.

I think in the future we would want to stream compilation messages in a more machine readable format.